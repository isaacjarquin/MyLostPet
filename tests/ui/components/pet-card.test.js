import "react-native"
import React from "react"
import PetCard from "../../../components/pet-card"

import renderer from "react-test-renderer"

it("renders correctly", () => {
    const navigation = {
        state: {
            params: {
                pet: {
                    kind: "kind",
                    info: "info",
                    image: "image",
                    location: "location",
                    date: "date",
                    breed: "breed"
                }
            }
        }
    }

    const tree = renderer.create(
        <PetCard navigation={navigation} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
})
