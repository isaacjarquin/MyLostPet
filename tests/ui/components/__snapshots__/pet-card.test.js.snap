// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "backgroundColor": "white",
          "flex": 1,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
              undefined,
            ]
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#43484d",
                      "fontSize": 17.5,
                      "fontWeight": "bold",
                      "marginBottom": 15,
                      "textAlign": "center",
                    },
                    Object {
                      "marginTop": 15,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              kind, de raza breed
            </Text>
          </View>
          <View
            style={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "height": 150,
                    "width": null,
                  },
                  Object {
                    "height": 300,
                  },
                ]
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "imageupload/w_300,h_340,c_fill,g_southundefined",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "backgroundColor": "rgba(0, 0, 0, 0.2)",
                      "bottom": 0,
                      "flex": 1,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 22.5,
                          "fontWeight": "800",
                          "marginBottom": 8,
                        },
                        undefined,
                      ],
                    ]
                  }
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 16.25,
                          "fontWeight": "400",
                          "marginBottom": 8,
                        },
                        undefined,
                      ],
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "padding": 10,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                    "marginBottom": 10,
                  }
                }
              >
                Encontrado en 
                location
                , el 
                date
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "lineHeight": 25,
                    "marginBottom": 20,
                  }
                }
              >
                info
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#F7F7F7",
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#333333",
              "flexDirection": "row",
              "opacity": 0.8,
              "padding": 25,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
                "fontSize": 18,
                "marginLeft": "35%",
              }
            }
          >
            Contactar
          </Text>
        </View>
        <Modal
          animationType="none"
          hardwareAccelerated={false}
          onModalHide={[Function]}
          onRequestClose={[Function]}
          transparent={true}
          visible={false}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "black",
                "bottom": 0,
                "height": 1334,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 750,
              }
            }
            testID={undefined}
          />
          <View
            collapsable={undefined}
            onModalHide={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "backgroundColor": "#333333",
                "borderTopLeftRadius": 3,
                "borderTopRightRadius": 3,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "margin": 37.5,
                "marginTop": 270,
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "grey",
                        "fontSize": 30,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        undefined,
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View>
              <TextInput
                borderColor="grey"
                onChangeText={[Function]}
                placeholder="Nombre"
                placeholderTextColor="black"
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#333333",
                    "borderWidth": 1,
                    "color": "black",
                    "fontSize": 16,
                    "height": 60,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  }
                }
                value=""
              />
              <TextInput
                borderColor="grey"
                onChangeText={[Function]}
                placeholder="Correo"
                placeholderTextColor="black"
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#333333",
                    "borderWidth": 1,
                    "color": "black",
                    "fontSize": 16,
                    "height": 60,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  }
                }
                value=""
              />
              <TextInput
                borderColor="grey"
                onChangeText={[Function]}
                placeholder="Número de teléfono"
                placeholderTextColor="black"
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#333333",
                    "borderWidth": 1,
                    "color": "black",
                    "fontSize": 16,
                    "height": 60,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  }
                }
                value=""
              />
              <TextInput
                borderColor="grey"
                multiline={true}
                numberOfLines={4}
                onChangeText={[Function]}
                placeholder="Información personal"
                placeholderTextColor="black"
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#333333",
                    "borderWidth": 1,
                    "color": "black",
                    "fontSize": 16,
                    "height": 100,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  }
                }
                value=""
              />
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "grey",
                    "flexDirection": "row",
                    "opacity": 1,
                    "padding": 20,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 25,
                      "marginLeft": "15%",
                      "marginRight": 10,
                      "width": 25,
                    }
                  }
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    }
                  }
                >
                  Enviar mis datos
                </Text>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
